modules/javafx.controls/src/test/java/test/javafx/scene/control/TreeViewTest.java

```diff
@@ -2343,6 +2344,98 @@ private void test_rt_37853(boolean replaceRoot) {
         sl.dispose();
     }
 
+    @Test
+    public void testTreeViewRemainsFocusedAfterEditCancel() {
+        treeView.setCellFactory(TextFieldTreeCell.forTreeView());
+        treeView.setEditable(true);
+        treeView.setRoot(new TreeItem<>("Root"));
+        treeView.getRoot().setExpanded(true);
+        treeView.setShowRoot(false);
+
+        TreeItem<String> item = new TreeItem<>("John");
+        treeView.getRoot().getChildren().add(item);
+
+        StageLoader sl = new StageLoader(new Button(), treeView);
+        treeView.requestFocus();
+        assertTrue(treeView.isFocused());
+
+        VirtualFlowTestUtils.BLOCK_STAGE_LOADER_DISPOSE = true;
```

Comment:
This should not be needed since you are creating a `stageLoader` above. This seems to be only used when no `stageLoader` was created before

Reply:
Thanks @Maran23, I've removed it now from the added tests.

Reply:
> I wish there was a more reliable way of implementing getCell() without resorting to undocumented flags like BLOCK_STAGE_LOADER_DISPOSE.

There is, just add the virtual container to a Stage. Since the cells are only created inside the skin, which is only created when the component is inside the scene tree.